src/lib/components/quiz/quiz.svelte

```diff
@@ -0,0 +1,357 @@
+<script lang="ts">
+  import { QuizData, QuizStorage, TeamMatch } from '$lib/quiz';
+  import { AcmPath, acmAlgo, acmDev, acmDesign, acmAI, acmGeneral } from '$lib/constants/acm-paths';
+  import { onMount } from 'svelte';
+  import ProgressBar from '$lib/components/quiz/progress-bar.svelte';
+  import LeftArrow from '$lib/components/icons/left-arrow.svelte';
+  import RightArrow from '$lib/components/icons/right-arrow.svelte';
+  import MoreInfo from './more-info.svelte';
+  import AcmPaths from '../index/acm-paths.svelte';
+
+  export let data: QuizData;
+
+  const TEAMS = {
+    [TeamMatch.AI]: acmAI,
+    [TeamMatch.DEV]: acmDev,
+    [TeamMatch.DESIGN]: acmDesign,
+    [TeamMatch.ALGO]: acmAlgo,
+  };
+  const TEAMLESS = { [TeamMatch.TEAMLESS]: acmGeneral };
+  //  [acmAlgo.title, acmDev.title, acmDesign.title, acmAI.title]
+  let index = 0;
+  let responses: TeamMatch[] = [];
+  let answeredAllQuestions = false;
+  // local storage stuff
+  let quizStorage: QuizStorage | undefined;
+  $: {
+    answeredAllQuestions =
+      responses.length === data.questions.length && !responses.includes(undefined);
+    // Updates local storage whenever `responses` changes.
+    quizStorage && quizStorage.setResponses(responses);
+  }
+  let showResults = false;
+  let showMoreInfo = false;
+  let showTeam: AcmPath;
+  $: talliedResponses = responses.reduce((tallies, match) => {
+    if (tallies[match]) tallies[match]++;
+    else tallies[match] = 1;
+    return tallies;
+  }, {} as Record<TeamMatch, number>);
+  $: match = (Object.entries(talliedResponses)
+    .sort(([, a], [, b]) => b - a)
+    ?.shift()
+    ?.shift() ?? TeamMatch.TEAMLESS) as TeamMatch;
+
+  function goLeft() {
+    if (index > 0) index--;
+  }
+
+  function goRight() {
+    if (index < data.questions.length - 1) {
+      index++;
+    }
+    if (index === data.questions.length) {
+      // need to display text at the bottom that says:
+      console.log('please answer all questions');
+    }
+  }
+
+  function recordAnswer(match: TeamMatch) {
+    responses[index] = match;
+    goRight();
+  }
+
+  function submitResponses() {
+    // show results and hide the question
+    showResults = true;
+  }
+
+  function restartQuiz() {
+    responses = [];
+    index = 0;
+    showResults = false;
+    quizStorage.clearResponses();
+  }
+
+  function showTeamDetails(currentTeam: AcmPath) {
+    showMoreInfo = true;
+    showTeam = currentTeam;
+  }
+
+  function goBackToResults() {
+    showMoreInfo = false;
+  }
+
+  onMount(() => {
+    // hypothetically change this to QuizStorage.init()
+    quizStorage = new QuizStorage();
```

Comment:
Load the visitor's previous answers if they exist in local storage.
```suggestion
    quizStorage = new QuizStorage();
    responses = quizStorage.getResponses();
```

Reply:
Done